src/page/util.rs

```diff
@@ -260,33 +284,123 @@ where
         }
     }
 
-    let mut result: Vec<IndexData<T>> = vec![];
+    let mut result: Vec<IndexValue<T>> = vec![];
+    for interval in intervals.iter() {
+        for index in interval.0..=interval.1 {
+            let mut index_records = parse_index_page::<T, PAGE_SIZE>(file, index as u32)?;
+            result.append(&mut index_records);
+        }
+    }
+
+    Ok(result)
+}
+
+pub fn read_index_pages<T, const PAGE_SIZE: usize>(
+    file: &mut std::fs::File,
+    intervals: &Vec<Interval>,
+) -> eyre::Result<Vec<IndexValue<T>>>
+where
+    T: Archive,
+    <T as rkyv::Archive>::Archived: rkyv::Deserialize<T, HighDeserializer<rkyv::rancor::Error>>,
+{
+    let mut result: Vec<IndexValue<T>> = vec![];
     for interval in intervals.iter() {
-        for index in interval.0..interval.1 {
-            let index_page = parse_index_page::<T, PAGE_SIZE>(file, index as u32)?;
-            result.push(index_page.inner);
+        for index in interval.0..=interval.1 {
```

Comment:
I'm a bit doubt when I see nested loops. For improving performance I suppose we can merge intervals to reduce amount of iterations.